pkg/controllers/provisioningv2/rke2/common.go

```diff
@@ -403,3 +406,84 @@ func SortedKeys(m map[string]interface{}) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+type CAPIClusterGetter interface {
+	Get(namespace, name string) (*capi.Cluster, error)
+}
+
+type CAPIMachineGetter interface {
+	Get(namespace, name string) (*capi.Machine, error)
+}
+
+var errNilObject = errors.New("cannot get capi cluster for nil object")
+
+// GetCAPIClusterFromLabel takes a runtime.Object and will attempt to find the label denoting which capi cluster it
+// belongs to.
+// If the object is nil, it cannot access to object or type metas, or the label is not present, it returns an error.
+// If the object has the expected label, it will return the capi cluster object.
+func GetCAPIClusterFromLabel(obj runtime.Object, getter CAPIClusterGetter) (*capi.Cluster, error) {
```

Comment:
What is the reason for passing in a getter?

Reply:
Like, is there a legitimate reason where the getter will be different? Or is this future-proofing?